src/pages/History/index.tsx

```diff
@@ -16,32 +22,59 @@ export const HistoryUi: React.FC<IHistoryUiProps> = ({
   showOnlyMyTxns,
   setShowOnlyMyTxns,
   transactions,
+  walletConnected,
+  isFetching,
+  total,
+  limit,
+  setCurrentPage,
 }) => (
   <div>
     <Filters
       eventType={eventType}
       setEventType={setEventType}
       showOnlyMyTxns={showOnlyMyTxns}
       setShowOnlyMyTxns={setShowOnlyMyTxns}
+      walletConnected={walletConnected}
     />
-    <HistoryTable transactions={transactions} />
+    <HistoryTable transactions={transactions} isFetching={isFetching} />
+    {total && (
+      <Pagination
+        itemsCount={total}
+        onChange={(nextIndex: number) => {
+          setCurrentPage(nextIndex + 1);
```

Comment:
As mentioned previously, I think the actual page number should reflect the index of the page in the array.
We're only changing the page number visually for users, so in my opinion this should be done on the UI inside `Pagination` directly (line 93 of this file: `src/components/v2/Pagination/index.tsx`)

Reply:
the backend pagination index starts at 1 which is why this is required. Following up in slack